trade_portal/trade_portal/oa_verify/services.py

```diff
@@ -197,3 +203,139 @@ def _parse_attachments(self, data):
                 }
             )
         return attachments
+
+
+class PdfVerificationService:
+    """
+    Service to extract supported QR codes from provided PDF
+
+    https://github.com/gs-gs/ha-igl-project/issues/54
+    """
+
+    def __init__(self, pdf_file: bytes):
+        self._pdf_binary = pdf_file
+
+    def get_valid_qrcodes(self):
+        """
+        For the PDF with which this service has been initialized
+        Tries to parse it
+        Retrieving all images and parsing them as QR codes
+        And if parsed - verify QR code format to be one of supported ones
+        And return the text from all the supported QR codes
+
+        Seems to handle scanned PDFs well, but real usage will give us a lot of complex PDFs which
```

Comment:
would it not be simpler to treat all documents as scanned PDFs? 

i.e. forget about digging into the structure of the PDF, just treat it as an image as if it were being read by camera.  

- for each page in pdf, convert to image, feed to pyzbar decode?

Reply:
it's like losing information to me before even parsing it; bit in case of no QR codes read it could be helpful (as a final step)